src/python/grapl-template-generator/grapl_template_generator/grapl-templates/rust-grpc-service/{{cookiecutter.project_slug}}/src/server.rs

```diff
@@ -0,0 +1,63 @@
+use tonic::{transport::Server, Request, Response, Status};
+
+use crate::{{cookiecutter.snake_project_name}}::{{cookiecutter.service_name}}Request;
+pub use crate::{{cookiecutter.snake_project_name}}::{{cookiecutter.service_name}}Response;
+pub use crate::{{cookiecutter.snake_project_name}}::{{cookiecutter.snake_project_name}}_rpc_server::{{cookiecutter.service_name}}Rpc;
+pub use crate::{{cookiecutter.snake_project_name}}::{{cookiecutter.snake_project_name}}_rpc_server::{{cookiecutter.service_name}}RpcServer;
+
+#[derive(Default)]
+pub struct {{cookiecutter.service_name}} {}
+
+#[tonic::async_trait]
+impl {{cookiecutter.service_name}}Rpc for {{cookiecutter.service_name}} {
+
+    #[tracing::instrument(
+        source_addr = request.remote_addr(),
+        client_id = request.get_ref().request_meta.client_id,
+        skip(self, request),
+    )]
+    async fn handle_request(
+        &self,
+        request: Request<{{cookiecutter.service_name}}Request>,
+    ) -> Result<Response<{{cookiecutter.service_name}}Response>, Status> {
+        let start = quanta::Instant::now();
+
+        print!("This is here to shush an unused-variable warning: {:?}", request);
+
+        let reply = {{cookiecutter.service_name}}Response {
+
+        };
+
+        let delta = quanta::Instant::now().duration_since(start);
+        metrics::histogram!("request_ns", delta);
+
+        Ok(Response::new(reply))
+    }
+}
+
+pub async fn exec_service()  -> Result<(), Box<dyn std::error::Error>> {
+    let (mut health_reporter, health_service) = tonic_health::server::health_reporter();
+    health_reporter
+    .set_serving::<{{cookiecutter.service_name}}RpcServer<{{cookiecutter.service_name}}>>()
+    .await;
+
+    let addr = "[::1]:50051".parse().unwrap();
```

Comment:
(where does this port even come from)

Reply:
I've done this, except in lib.rs (since it seems odd for the tests to pull stuff from `main`)